src/components/profileMenu.jsx

```diff
@@ -12,31 +12,49 @@ import {
 import { BsThreeDotsVertical } from 'react-icons/bs';
 import { CopyToClipboard } from 'react-copy-to-clipboard';
 import { useInjectedProvider } from '../contexts/InjectedProviderContext';
-import { useOverlay } from '../contexts/OverlayContext';
-import { useUser } from '../contexts/UserContext';
-import { useDao } from '../contexts/DaoContext';
-import { createPoll } from '../services/pollService';
-import { MolochService } from '../services/molochService';
+import { useFormModal, useOverlay } from '../contexts/OverlayContext';
+import { useDaoMember } from '../contexts/DaoMemberContext';
+import { useTX } from '../contexts/TXContext';
+import { TX } from '../data/contractTX';
+import { CORE_FORMS, FORM } from '../data/forms';
 import { daoConnectedAndSameChain } from '../utils/general';
+import { createContract } from '../utils/contract';
+import { LOCAL_ABI } from '../utils/abi';
 
 const ProfileMenu = ({ member }) => {
   const toast = useToast();
   const { address, injectedChain, injectedProvider } = useInjectedProvider();
+  const { openFormModal } = useFormModal();
   const { daochain, daoid } = useParams();
-  const { daoOverview } = useDao();
-  const {
-    setGenericModal,
-    errorToast,
-    successToast,
-    setTxInfoModal,
-  } = useOverlay();
-  const { cachePoll, resolvePoll } = useUser();
-  const history = useHistory();
-
-  const handleGuildkickClick = () => {
-    history.push(
-      `/dao/${daochain}/${daoid}/proposals/new/guildkick?applicant=${member.memberAddress}`,
-    );
+  const { daoMember } = useDaoMember();
+  const { errorToast } = useOverlay();
+  const { submitTransaction } = useTX();
+
+  const [canRageQuit, setCanRageQuit] = useState(false);
+
+  const handleGuildKickClick = () => {
+    openFormModal({
+      lego: {
+        ...FORM.GUILDKICK,
+        localValues: { memberAddress: member.memberAddress },
+      },
+    });
+  };
+
+  const handleRageQuitClick = () => {
+    openFormModal({
+      lego: {
```

Comment:
no need to create a new object on this or the next one - only need to do that if we are overriding a lego property or adding localValues like in guildkick.

```
    lego: CORE_FORMS.RAGE_QUIT
```